[CMake] Support USE(SKIA) build on PlayStation

donny-dont opened the pull request:
#### b2384277ae56839abfca022f4c70fdf6f9b38cc3
<pre>
[CMake] Support USE(SKIA) build on PlayStation
<a href="https://bugs.webkit.org/show_bug.cgi?id=269250">https://bugs.webkit.org/show_bug.cgi?id=269250</a>

Reviewed by Fujii Hironori.

Handle the `USE(SKIA)` case within the PlayStation build of WebKit. Modify the
build to handle both `USE(CAIRO)` and `USE(SKIA)`. Guard more code with
`USE(CAIRO)`.

Cairo has to be used when building the MiniBrowser at this time so also support
that.

Building Skia itself in a cross platform way will be in separate patches.

* Source/WebCore/PlatformPlayStation.cmake:
* Source/WebKit/GPUProcess/EntryPoint/playstation/GPUProcessMain.cpp:
(main):
* Source/WebKit/PlatformPlayStation.cmake:
* Source/WebKit/UIProcess/API/C/playstation/WKPagePrivatePlayStation.cpp:
(WKPagePaint):
* Source/WebKit/WebProcess/EntryPoint/playstation/WebProcessMain.cpp:
(main):
* Source/cmake/OptionsPlayStation.cmake:
* Tools/MiniBrowser/playstation/CMakeLists.txt:
* Tools/MiniBrowser/playstation/WebViewWindow.cpp:
(WebViewWindow::setSize):
(WebViewWindow::paintSelf):
* Tools/MiniBrowser/playstation/main.cpp:
(initialize):
* Tools/TestWebKitAPI/playstation/main.cpp:
(main):
* Tools/WebKitTestRunner/PlatformPlayStation.cmake:

Canonical link: <a href="https://commits.webkit.org/274675@main">https://commits.webkit.org/274675@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/bd7058403a1d2928eb71128b6c2e3300adf9be51

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/39752 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/18731 "Built successfully") | [  ~~🛠 mac~~](https://ews-build.webkit.org/#/builders/14/builds/42108 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 wpe~~](https://ews-build.webkit.org/#/builders/5/builds/42287 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/32/builds/35654 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/42059 "Passed tests") | [  ~~🛠 ios-sim~~](https://ews-build.webkit.org/#/builders/49/builds/21637 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/16060 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/5/builds/42287 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/40326 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/49/builds/21637 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/14/builds/42108 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-wpe~~](https://ews-build.webkit.org/#/builders/5/builds/42287 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/49/builds/21637 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/14/builds/42108 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 gtk~~](https://ews-build.webkit.org/#/builders/2/builds/43564 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [  ~~🛠 🧪 jsc~~](https://ews-build.webkit.org/#/builders/20/builds/33205 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/49/builds/21637 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/14/builds/42108 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/2/builds/43564 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/39378 "Built successfully and passed tests") | [  ~~🛠 tv~~](https://ews-build.webkit.org/#/builders/44/builds/14567 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/50/builds/11994 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/2/builds/43564 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/46/builds/16170 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk2-stress~~](https://ews-build.webkit.org/#/builders/14/builds/42108 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/46386 "Built successfully") | 
| | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/43/builds/16234 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | [  ~~🧪 jsc-armv7-tests~~](https://ews-build.webkit.org/#/builders/25/builds/9538 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/5221 "Built successfully and passed tests") | [  ~~🛠 watch-sim~~](https://ews-build.webkit.org/#/builders/45/builds/15830 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
<!--EWS-Status-Bubble-End-->

fujii commented on Source/WebKit/Shared/API/c/cairo/WKImageCairo.cpp:
```diff
@@ -49,6 +49,9 @@ cairo_surface_t* WKImageCreateCairoSurface(WKImageRef imageRef)
 {
+#if USE(SKIA)
+    return nullptr;
+#else
```
If you break an API, it doesn't make sense to keep `WKImageCreateFromCairoSurface` doing nothing.

donny-dont committed:
[CMake] Support USE(SKIA) build on PlayStation
https://bugs.webkit.org/show_bug.cgi?id=269250

Reviewed by Fujii Hironori.

Handle the `USE(SKIA)` case within the PlayStation build of WebKit. Modify the
build to handle both `USE(CAIRO)` and `USE(SKIA)`. Guard more code with
`USE(CAIRO)`.

Cairo has to be used when building the MiniBrowser at this time so also support
that.

Building Skia itself in a cross platform way will be in separate patches.

* Source/WebCore/PlatformPlayStation.cmake:
* Source/WebKit/GPUProcess/EntryPoint/playstation/GPUProcessMain.cpp:
(main):
* Source/WebKit/PlatformPlayStation.cmake:
* Source/WebKit/UIProcess/API/C/playstation/WKPagePrivatePlayStation.cpp:
(WKPagePaint):
* Source/WebKit/WebProcess/EntryPoint/playstation/WebProcessMain.cpp:
(main):
* Source/cmake/OptionsPlayStation.cmake:
* Tools/MiniBrowser/playstation/CMakeLists.txt:
* Tools/MiniBrowser/playstation/WebViewWindow.cpp:
(WebViewWindow::setSize):
(WebViewWindow::paintSelf):
* Tools/MiniBrowser/playstation/main.cpp:
(initialize):
* Tools/TestWebKitAPI/playstation/main.cpp:
(main):
* Tools/WebKitTestRunner/PlatformPlayStation.cmake:

Canonical link: https://commits.webkit.org/274675@main
